fdbserver/storageserver.actor.cpp

```diff
@@ -955,7 +957,13 @@ ACTOR Future<Void> watchValue_impl( StorageServer* data, WatchValueRequest req )
 				++data->numWatches;
 				data->watchBytes += ( req.key.expectedSize() + req.value.expectedSize() + 1000 );
 				try {
-					wait( watchFuture );
+					if(latest < minVersion) {
+						// If the version we read is less than minVersion, then we may fail to be notified of any changes that occur up to or including minVersion
+						// To prevent that, we'll check the key again once the version reaches our minVersion
+						watchFuture = watchFuture || data->version.whenAtLeast(minVersion);
+					}
+					wait(watchFuture);
+					wait(data->version.whenAtLeast(data->data().latestVersion));
```

Comment:
Why do we need this line?

Reply:
This is roughly equivalent to what we had before, but instead of waiting inside the get value call, we do it here instead (but without a potential future version). We have to do it before the call now because otherwise the version we pass in will be older than it was before and not correct. I think it’s the case that this wait will usually be pretty quick because the version is updated shortly after applying it and triggering the watch.